scripts/perf/scheduler_dag_execution_timing.py

```diff
@@ -0,0 +1,221 @@
+#!/usr/bin/env python3
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import gc
+import os
+import statistics
+import time
+
+import click
+
+
+class ShortCircutExecutorMixin:
+    def __init__(self, stop_when_these_completed):
+        super().__init__()
```

Comment:
Do we need a super call here as we are not subclassing anything?

Reply:
Yes, because it's a MixIn.

Reply:
Used like this:

https://github.com/apache/airflow/pull/7537/files#diff-fda366d49e19f54d95380db869c369e1R68-R71

```python
    Executor = MockExecutor

    class ShortCircutExecutor(ShortCircutExecutorMixin, Executor):
        pass
```